internal/stream_metadata.go

```diff
@@ -32,12 +33,23 @@ func GetBackupStreamFetcher(backup Backup) (StreamFetcher, error) {
 		return nil, err
 	}
 
+	limitedFileSize := false
+	if metadata.Type == LimitedFileSizeBackup {
+		limitedFileSize = true
+	}
+
 	switch metadata.Type {
+	case LimitedFileSizeBackup:
```

Comment:
It looks like thees two cases differ only in the last argument. May be join them together ?  And get rid of temporary  `limitedFileSize` variable